src/hooks/useDimensions.ts

```diff
@@ -0,0 +1,29 @@
+import { MutableRefObject, useLayoutEffect, useMemo, useState } from "react";
+
+interface DimensionState {
+  width: number;
+  height: number;
+}
+
+export const useDimensions = (ref: MutableRefObject<HTMLElement>) => {
+  const [state, setState] = useState<DimensionState>({ width: 0, height: 0 });
+
+  const observer = useMemo(
```

Comment:
This useMemo isn't necessary you can declare this in a singular useEffect and have it handle the disconnect state.  `useIntersectionObserver` should behave similarly.
https://github.com/evergreen-ci/spruce/blob/c4b2c563e1f49d71e3a509122c594568025aa65a/src/hooks/useIntersectionObserver/index.ts